src/macfloat.c

```diff
@@ -70,6 +70,10 @@ static void PrintMacfloatToBuf(char *buf, size_t bufsize, Double val, int precis
     }
     else {
         snprintf(buf, bufsize, "%.*" PRINTFFORMAT, precision, val);
+        if(!strchr(buf, '.'))
+        {
+          strcat(buf, ".");
```

Comment:
Please do not use `strcat`, it is a potential source of buffer overflows. use `strlcat` or `strxcat` instead.

Reply:
Changed it.